### PR TITLE
Login Page

### LINKED ISSUE #23: Login page

### PR DESCRIPTION
**Pull request by zeetec20**
fix #23

### CODE REVIEW

File: app/(auth)/login/page.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
+'use client'
+import Head from 'next/head'
+import Image from 'next/image'
+import { FormEvent, useState } from 'react'
+import { FaFacebook, FaTwitter } from 'react-icons/fa'
+import Logo from 'src/assets/images/logo.png'
```
Comment (kalwabed) [inline app/(auth)/login/page.tsx:6]:
```suggestion
import Logo from '~/assets/images/logo.png'
```

File: app/(auth)/login/page.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  const onSubmit = (e: FormEvent) => {
+    e.preventDefault()
+    setIsSubmit(true)
+  }
+
+  return (
+    <div>
+      <Head>
+        <title>Login</title>
+      </Head>
```
Comment (kalwabed) [inline app/(auth)/login/page.tsx:23]: Please use separate `head` files.

Reference: https://beta.nextjs.org/docs/api-reference/file-conventions/head

File: app/(auth)/login/page.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+      <div className="flex min-h-screen w-screen flex-col items-center justify-center bg-c-white-broken">
+        <div className="mt-8 mb-9 flex items-center justify-center overflow-hidden rounded-full bg-white p-3 shadow-md">
+          <Image src={Logo} alt="logo site" width={75} height={75} />
+        </div>
+
+        <form
+          className={`bg-w group h-min w-[22%] rounded border-t-2 border-t-c-purple bg-white shadow-sm ${
+            isSubmit ? 'useValidation' : ''
+          }`}
```
Comment (kalwabed) [inline app/(auth)/login/page.tsx:33]: Please use `clsxm()` from utils.

File: src/components/form/label.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { DetailedHTMLProps, LabelHTMLAttributes } from 'react'
+
+const Label = (props: DetailedHTMLProps<LabelHTMLAttributes<HTMLLabelElement>, HTMLLabelElement>) => {
+  const className = `
+        text-xs 
+        text-c-black
+        ${props.className ?? ''}
+    `
+  const propsMerge = { ...props, className }
```
Comment (kalwabed) [inline src/components/form/label.tsx:9]: Please use `clsxm` util to compose classnames.

File: src/components/form/text-input.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      rounded-[4px]
+      block 
+      w-full 
+      p-2.5
+      outline-none
+      focus:border-c-light-purple
+      text-c-black
+      ${props.className ?? ''}
+  `
+  const propsMerge = { ...props, className }
```
Comment (kalwabed) [inline src/components/form/text-input.tsx:19]: Please use `clsxm` util to compose classnames.

File: tailwind.config.js [hunk 6]
```diff
@@ -11,6 +11,16 @@ module.exports = {
...
       },
+      colors: {
+        'c-black': '#34395e',
+        'c-white': '#f9f9f9',
+        'c-white-broken': '#FAFDFB',
+        'c-light-purple': '#95a0f4',
+        'c-light-purple-2': '#acb5f6',
+        'c-purple': '#6777EF',
+        'c-white-purple': '#fdfdff',
+        'c-white-purple-2': '#e4e6fc',
```
Comment (kalwabed) [inline tailwind.config.js:22]: Please be more verbose. Change `c` to `custom`.

**Review by kalwabed — CHANGES_REQUESTED** [review]

File: app/(auth)/login/page.tsx (hunk 1 above)
Comment (zeetec20) [reply app/(auth)/login/page.tsx:6]: Waiting #42

File: src/components/ui/form/text-input.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import { DetailedHTMLProps, InputHTMLAttributes } from 'react'
+
+import clsxm from '~/utils/clsxm'
+
+const TextInput = (props: DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>) => {
+  const className = clsxm(
+    'mt-2 bg-c-white-purple border border-c-white-purple-2 text-sm rounded-[4px] block w-full p-2.5 outline-none focus:border-c-light-purple text-c-black',
```
Comment (kalwabed) [inline src/components/ui/form/text-input.tsx:7]: I think you forgot to change from `c` to `custom`, for example: `bg-c-white-purple`.

File: src/components/ui/form/label.tsx [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import { DetailedHTMLProps, LabelHTMLAttributes } from 'react'
+
+import clsxm from '~/utils/clsxm'
+
+const Label = (props: DetailedHTMLProps<LabelHTMLAttributes<HTMLLabelElement>, HTMLLabelElement>) => {
+  const className = clsxm('text-xs text-c-black', props.className)
```
Comment (kalwabed) [inline src/components/ui/form/label.tsx:6]: I think you forgot to change from `c` to `custom`.

File: app/(auth)/login/page.tsx [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+const LoginPage = () => {
+  const [isSubmit, setIsSubmit] = useState(false)
+
+  const onSubmit = (e: FormEvent) => {
+    e.preventDefault()
+    setIsSubmit(true)
+  }
+
+  return (
+    <div className="bg-c-white-broken flex min-h-screen w-screen flex-col items-center justify-center">
```
Comment (kalwabed) [inline app/(auth)/login/page.tsx:20]: I think you forgot to change from `c` to `custom`

**Review by kalwabed — CHANGES_REQUESTED** [review]

**Review by kalwabed — APPROVED** [review]
LGTM :100:

**Pull request merged**